aptos-move/framework/aptos-stdlib/sources/simple_map.move

```diff
@@ -134,6 +134,14 @@ module aptos_std::simple_map {
         (key, value)
     }
 
+    /// Get the key at a given index in the underlying vector.
+    public fun key_at_idx<Key: store, Value: store>(
```

Comment:
It seems a bit wrong to expose the fact that our SimpleMap is just a vector.

Reply:
I could change the comment? Of course some approach where there is a "for each" but with references would be ideal, but it doesn't seem like that's possible.

Reply:
I think Gerben's point is we instead should have sth like an iterator but the way to use that iterator would be very cumbersome.

Reply:
What would it look like to use this iterator? Could you share a hypothetical code snippet?

Reply:
There is no ideal implementation now as move struct cannot have reference inside. What I can think of that may be close to idea is to have a inline function like this:
```
inline fun for_each<K, V>(m: &SimpleMap, f: |&Element|) {
}
```
and expose `Element`. But IIRC, there is some issues with inline functions.
So my suggestion would generating all the keys.